R/get_analysis_date.R

```diff
@@ -231,16 +235,14 @@ get_analysis_date <- function(
   # 2b: Reach targeted events per sub-population
   if (!all(is.na(target_event_per_stratum))) {
     stratum <- unique(data$stratum)
-    cut_date2b <- lapply(
-      seq_along(target_event_per_stratum),
-      function(x) {
-        get_cut_date_by_event(data %>% dplyr::filter(stratum == stratum[x]),
-          event = target_event_per_stratum[x]
-        )
-      }
-    ) %>%
-      unlist() %>%
-      max()
+    cut_date2b <- vector(mode = "list", length = length(stratum))
+    for (i in seq_along(target_event_per_stratum)) {
```

Comment:
Hi John, sorry for the not timely review, I was on a flight last Tuesday and Wednesday. May I know if the change from `lappy` to `for`-loop, is because `lappy` is slow?

Reply:
No, I didn't remove `lapply()` for speed reasons. `lapply()` is perfectly fast.

The main motivation was scoping (as @nanxstats described in https://github.com/Merck/simtrial/pull/122#pullrequestreview-1711786166. The anonymous function only had a single argument `x`, but it accessed multiple variables outside its scope (`data`, `stratum`, `target_event_per_stratum`). One option would have been to create a new internal function with all the required arguments. But since the code is so minimal (filtering followed by passing to `get_cut_date_by_event()`), I instead decided to put in a `for` loop (which doesn't have the scoping issues that an anonymous function has).